File: data structures and algorithms/java/06_010.java
// Iterative inorder example
public List<Integer> inorderIterative(TreeNode root) {
   List<Integer> result = new ArrayList<>();
   Stack<TreeNode> stack = new Stack<>();
   TreeNode current = root;
   
   while (current != null || !stack.isEmpty()) {
       while (current != null) {
           stack.push(current);
           current = current.left;
       }
       current = stack.pop();
       result.add(current.val);
       current = current.right;
   }
   
   return result;
}

File: data structures and algorithms/cs/06_010.cs
// Iterative inorder example
public List<int> InorderIterative(TreeNode root)
{
    List<int> result = new List<int>();
    Stack<TreeNode> stack = new Stack<TreeNode>();
    TreeNode current = root;

    while (current != null || stack.Count > 0)
    {
        while (current != null)
        {
            stack.Push(current);
            current = current.Left;
        }
        current = stack.Pop();
        result.Add(current.Value);
        current = current.Right;
    }

    return result;
}
